cli/cmd/generate_aws.go

```diff
@@ -595,677 +671,580 @@ func validateAwsAssumeRole(val interface{}) error {
+			Icon:     IconCloudTrail,
+			Prompt:   &survey.Confirm{Message: QuestionCloudtrailSqsEnableEncryption, Default: config.SqsEncryptionEnabled},
 			Response: &config.SqsEncryptionEnabled,
-			Checks:   []*bool{&config.Cloudtrail},
 		},
-		// Allow the user to set the SSE Key ARN if required
 		{
-			Prompt:   &survey.Input{Message: QuestionSqsEncryptionKeyArn, Default: config.SqsEncryptionKeyArn},
+			Icon:     IconCloudTrail,
+			Prompt:   &survey.Input{Message: QuestionCloudtrailSqsEncryptionKeyArn, Default: config.SqsEncryptionKeyArn},
 			Response: &config.SqsEncryptionKeyArn,
 			Opts:     []survey.AskOpt{survey.WithValidator(validateOptionalAwsArnFormat)},
-			Checks:   []*bool{&config.Cloudtrail, &config.SqsEncryptionEnabled},
+			Checks:   []*bool{&config.SqsEncryptionEnabled},
 		},
-	}, config.Cloudtrail); err != nil {
+	}); err != nil {
 		return err
 	}
-
 	return nil
 }
 
-func existingCloudTrailDisabled(extraState *AwsGenerateCommandExtraState) *bool {
-	existingCloudTrailDisabled := !extraState.UseExistingCloudtrail
-	return &existingCloudTrailDisabled
-}
-
-func promptAwsExistingIamQuestions(config *aws.GenerateAwsTfConfigurationArgs) error {
+func promptCloudtrailIAMQuestions(config *aws.GenerateAwsTfConfigurationArgs) error {
 	// ensure struct is initialized
 	if config.ExistingIamRole == nil {
 		config.ExistingIamRole = &aws.ExistingIamRoleDetails{}
 	}
 
 	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
 		{
-			Prompt:   &survey.Input{Message: QuestionExistingIamRoleName, Default: config.ExistingIamRole.Name},
+			Icon:     IconCloudTrail,
+			Prompt:   &survey.Input{Message: QuestionCloudtrailExistingIamRoleName, Default: config.ExistingIamRole.Name},
 			Response: &config.ExistingIamRole.Name,
-			Opts:     []survey.AskOpt{survey.WithValidator(survey.Required)},
+			// Opts:     []survey.AskOpt{survey.WithValidator(survey.Required)},
+			Required: true,
 		},
 		{
-			Prompt:   &survey.Input{Message: QuestionExistingIamRoleArn, Default: config.ExistingIamRole.Arn},
+			Icon:     IconCloudTrail,
+			Prompt:   &survey.Input{Message: QuestionCloudtrailExistingIamRoleArn, Default: config.ExistingIamRole.Arn},
 			Response: &config.ExistingIamRole.Arn,
-			Opts:     []survey.AskOpt{survey.WithValidator(survey.Required), survey.WithValidator(validateAwsArnFormat)},
+			Opts:     []survey.AskOpt{survey.WithValidator(validateAwsArnFormat)},
+			Required: true,
 		},
 		{
-			Prompt:   &survey.Input{Message: QuestionExistingIamRoleExtID, Default: config.ExistingIamRole.ExternalId},
+			Icon:     IconCloudTrail,
+			Prompt:   &survey.Input{Message: QuestionCloudtrailExistingIamRoleExtID, Default: config.ExistingIamRole.ExternalId},
 			Response: &config.ExistingIamRole.ExternalId,
-			Opts:     []survey.AskOpt{survey.WithValidator(survey.Required)},
+			// Opts:     []survey.AskOpt{survey.WithValidator(survey.Required)},
```

Comment:
nit: does this commented out code need to be here

Reply:
Removed.